core/trino-main/src/main/java/io/trino/operator/DirectExchangeBuffer.java

```diff
@@ -58,6 +59,8 @@
 
     int getSpilledPageCount();
 
+    TDigest getBufferUtilization();
```

Comment:
I would move the `TDigest` to the `DirectExchangeClient` field and only expose the current utilization (e.g. `double getCurrentUtilization()`). That way only the`DirectExchangeClient`  has to worry about TDigest updates

Reply:
It is not possible. There is no term "utilization" in the sense of `DirectExchangeClient` - there is no capacity of client, there is a capacity of `buffer`.